app/controllers/foods_controller.rb

```diff
@@ -1,11 +1,38 @@
 class FoodsController < ApplicationController
-  def index; end
+  before_action :authenticate_user!
 
-  def show; end
+  def index
+    @user = current_user
+    @foods = Food.all
+  end
 
-  def new; end
+  def show
+    @food = Food.find(params[:id])
+  end
 
-  def create; end
+  def new
+    @food = current_user.foods.new
+  end
 
-  def destroy; end
+  def create
+    @food = current_user.foods.create(food_params)
+
+    if @food.save
+      redirect_to new_food_path
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @food = current_user.foods.find(params[:id])
+    @food.destroy
+    redirect_to new_food_path
```

Comment:
Same comment for this, it should redirect to the foods page and it should have some flash messages.

Reply:
Done also